config/config.go

```diff
@@ -136,19 +138,30 @@ const (
 )
 
 type Metrics struct {
-	Type   MetricsType `mapstructure:"type"`
-	Influx Influx      `mapstructure:"influx"`
+	Type       MetricsType       `mapstructure:"type"`
+	Influx     InfluxMetrics     `mapstructure:"influx"`
+	Prometheus PrometheusMetrics `mapstructure:"prometheus"`
 }
 
 func (cfg *Metrics) validateAndLog() {
-	log.Infof("config.metrics.type: %s", cfg.Type)
-	switch cfg.Type {
-	case MetricsNone:
-	case MetricsInflux:
-		cfg.Influx.validateAndLog()
-	default:
-		log.Fatalf(`invalid config.metrics.type: %s. It must be "none" or "influx"`, cfg.Type)
+	var err error
+	err = cfg.checkMetricsEnabled()
+
+	if err != nil {
+		log.Fatalf(err.Error())
+	}
+}
+
+func (cfg *Metrics) checkMetricsEnabled() error {
+	var err error
+	cfg.Influx.Enabled = (cfg.Type == MetricsInflux || cfg.Influx.Enabled)
+	cfg.Influx.Enabled = cfg.Influx.validateAndLogMetricsData()
```

Comment:
These two Influx lines look a bit strange to me. I think it's because we're using `cfg.Influx.Enabled` as implicit input and an error return channel, in addition to setting it within the method. Consider a flow like this:

```
func (cfg *Metrics) validate() {
  cfg.Influx.Enabled = (cfg.Type == MetricsInflux || cfg.Influx.Enabled)
  if (cfg.Influx.Enabled) {
    cfg.Influx.Enabled = cfg.Influx.validateAndLogMetricsData()
  }

  if (cfg.Prometheus.Enabled) {
    cfg.Influx.Enabled = cfg.Prometheus.validateAndLogMetricsData()
  }
}
```

Reply:
Done.